parity/main.rs

```diff
@@ -132,14 +146,50 @@ fn main_direct(force_can_restart: bool) -> i32 {
 	// increase max number of open files
 	raise_fd_limit();
 
-	let exit = Arc::new((Mutex::new((false, None)), Condvar::new()));
+	let exit = Arc::new((Mutex::new(ExitStatus {
+		panicking: false,
+		should_exit: false,
+		should_restart: false,
+		spec_name_override: None
+	}), Condvar::new()));
+
+	let exiting = Arc::new(AtomicBool::new(false));
```

Comment:
I think you should document here what you said in the comment of the PR.
It's not obvious why this exists and is not part of `ExitStatus`.